### ISSUE TITLE
Release NTL-trained models on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [ai4sd/number-token-loss#10]
Hi @jannisborn 🤗

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2411.02083.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models for instance),
you can also claim the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I noticed in your `number-token-loss-reproduction` repository that the trained checkpoints of your models are planned to be made available on the Hugging Face Hub soon.

Would you like to host these models on https://huggingface.co/models?
Hosting on Hugging Face will give you more visibility/enable better discoverability. We can add tags in the model cards so that people find the models easier, link it to the paper page, etc.

If you're down, leaving a guide [here](https://huggingface.co/docs/hub/models-uploading). If it's a custom PyTorch model, you can use the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin)
class which adds `from_pretrained` and `push_to_hub` to the model which lets you to upload the model and people to download and use models right away.
If you do not want this and directly want to upload model through UI or however you want, people can also use [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file).

After uploaded, we can also link the models to the paper page (read [here](https://huggingface.co/docs/hub/en/model-cards#linking-a-paper)) so people can discover your model.

You can also build a demo for your model on [Spaces](https://huggingface.co/spaces), we can provide you a ZeroGPU [grant](https://huggingface.co/docs/hub/en/spaces-gpus#community-gpu-grants), which gives you A100 GPUs for free.

Let me know if you're interested/need any guidance :)

Kind regards,

Niels

**Comment by jannisborn** [comment]
Hi @NielsRogge,

Thanks so much for reaching out and expressing interest! Honestly, we'd be thrilled to contribute some of our models to the HF hub 🤩  

Most of our models are standard HF models. Inference requires zero custom code, however, the training code requires a few extra lines. The main contribution of our paper is a regression-like loss that can be computed directly on the token head of a LLM and improves performance on numerical tasks. We would love to see this loss available natively in HF as this could really accelerate adoption by the community. We'd be excited to contribute models to the hub, but it would require a PR that contains the loss class. We're happy to work in this direction, so let me know what you think! 

Meanwhile, I've added linked the codebase, the project page and our [Streamlit demo on HF spaces](https://ibm.biz/ntl-demo) to the paper entry. The title/abstract itself is quite outdated though because it is from our initial preprint. The full paper was presented at ICML this year, are you able to update title/abstract based on the latest version from arxiv, see [here](https://ibm.biz/ntl-paper).

Thanks a lot for the support 🙏🏼 
Jannis

**Comment by NielsRogge** [comment]
Thanks a lot, I've pinged the team about the ability to update paper pages, that's not supported for now.

Thanks for making a demo and linking it with the paper!

**Comment by jannisborn** [comment]
Hi @NielsRogge thanks a lot for this. My actual question was whether a PR in the `transformers` package will be accepted if we add the loss there. It should be a couple of dozen of lines of custom code that is required for training. For inference we dont need anything. We'd be happy to work towards this but wanted to confirm with you first. 
Thanks a lot 🙏🏼

**Comment by jannisborn** [comment]
Could you be more specific on what "enough traction" means?
